imqsauth.go

```diff
@@ -234,6 +251,221 @@ func createDB(config *authaus.Config) bool {
+
+	// General info
+	// ------------
+
+	// Migration version
+	v, e := authaus.GetMigrationVersion(&ic.Config.Authaus.DB)
+	if e != nil {
+		fmt.Fprintf(w, "Could not read migration version...\n%v\n", e)
+	} else {
+		fmt.Fprintf(w, "Migration version:\t%v\n", v)
+	}
+
+	// Groups
+	groups, err := db.GetGroups()
+	if err == nil {
+		fmt.Fprintf(w, "Total nr of groups:\t%v\n", len(groups))
+	} else {
+		fmt.Fprintf(w, "Total nr of groups:\t%v\n", err)
+	}
+
+	// Users
+	users, err := ic.Central.GetAuthenticatorIdentities(authaus.GetIdentitiesFlagDeleted)
+	if err == nil {
+		fmt.Fprintf(w, "Total nr of users:\t%v\n", len(users))
+	} else {
+		fmt.Fprintf(w, "Total nr of users:\t%v\n", err)
+	}
+
+	// Non-archived users without sessions
+	tokens, e := ic.Central.GetAllTokens(false)
+	if e != nil {
+		fmt.Fprintf(w, "Could not retrieve tokens from session db\n")
+	} else {
+		fmt.Fprintf(w, "Active auth session tokens retrieved:\t%v\n", len(tokens))
+	}
+
+	oAuthTokenIDs, e := ic.Central.GetAllOAuthTokenIDs()
+	if e != nil {
+		fmt.Fprintf(w, "Could not retrieve tokens from session db\n")
+	} else {
+		fmt.Fprintf(w, "OAuth session tokens retrieved:\t%v\n", len(oAuthTokenIDs))
+	}
+
+	// detect any missing oauth sessions
+	nrSessionsWithOrphanedOAuthIds := findMissingOAuthTokens(oAuthTokenIDs, tokens)
+	fmt.Fprintf(w, "Nr of sessions with orphaned OAuth IDs:\t%v\n", nrSessionsWithOrphanedOAuthIds)
+
+	stats := map[string]int{}
+	stats[enabled] = 0
+	stats[admin] = 0
+	stats[archived] = 0
+	stats[unArchivedWithoutPermit] = 0
+
+	typeCounts := map[authaus.AuthUserType]*enabledTotal{
+		authaus.UserTypeDefault: {0, 0},
+		authaus.UserTypeLDAP:    {0, 0},
+		authaus.UserTypeOAuth:   {0, 0},
+		authaus.UserTypeMSAAD:   {0, 0},
+	}
+	missingGroups, errPerms2, _ := getUserStats(ic, users, stats, typeCounts, errorSet)
+	errFlag := ""
+	if errPerms2 {
+		errFlag = "*"
+	}
+	fmt.Fprintf(w, "Nr of enabled users:\t%v%v\n", stats[enabled], errFlag)
+	fmt.Fprintf(w, "Nr of admin users:\t%v%v\n", stats[admin], errFlag)
+	fmt.Fprintf(w, "Nr of archived users:\t%v\n", stats[archived])
+
+	fmt.Fprintf(w, "User type counts (archived):\n")
+	typeKeys := maps.Keys(typeCounts)
+	slices.SortFunc(typeKeys, func(a authaus.AuthUserType, b authaus.AuthUserType) int { return int(a) - int(b) })
+	for _, v := range typeKeys {
+		fmt.Fprintf(w, "\tType %v(%v):\t%v\t(%v)\n", authaus.AuthUserTypeStrings[v], v, typeCounts[v].enabled, typeCounts[v].total)
+	}
+
+	// Error checks
+	// ------------
+
+	// Permits check
+	fmt.Fprintf(w, "Permits:\n")
+	plist, err := ic.Central.GetPermits()
+	if err != nil {
+		errorSet[err.Error()] = err
+		fmt.Fprintf(w, "\tPermit list ERROR\n")
+	} else {
+		fmt.Fprintf(w, "\tPermit list ok:\t%v permits\n", len(plist))
+		fmt.Fprintf(w, "\t*Nr of non-archived users without permits:\t%v\n", stats[unArchivedWithoutPermit])
+		fmt.Fprintf(w, "\t*Nr of archived users without permits:\t%v\n", len(users)-len(plist)-stats[unArchivedWithoutPermit])
```

Comment:
These two printout are depend on `ic.Central.GetAuthenticatorIdentities(authaus.GetIdentitiesFlagDeleted)` to provide accurate information. I would rather not display this information if that function failed.